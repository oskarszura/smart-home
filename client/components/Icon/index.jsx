import React from 'react';
import PropTypes from 'prop-types';

const Icon = (props) => {
  const { type, className } = props;

  const classes = `c-icon c-icon--${type} ${className}`;

  return (<span className={classes} />);
};

Icon.propTypes = {
  className: PropTypes.string,
  type: PropTypes.string,
};

export default Icon;

