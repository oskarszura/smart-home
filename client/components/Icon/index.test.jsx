import React from 'react';
import { mount } from 'enzyme';
import Icon from './index';
import '../../../setupTest';

describe('Icon', () => {
  it('should render correctly', () => {
    const component = mount(<Icon />);

    expect(component).toMatchSnapshot();
  });
});
