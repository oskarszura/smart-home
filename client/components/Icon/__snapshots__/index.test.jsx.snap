// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render correctly 1`] = `
<Icon>
  <span
    className="c-icon c-icon--undefined undefined"
  />
</Icon>
`;
